<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from panda_mobile.urdf.xacro        | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="panda">
  <!--    Root world link-->
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="0.000 0.000 -1.500" xyz="0.004 1.087 0.000"/>
  </joint>
  <material name="dark_grey">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>
  <material name="light_grey">
    <color rgba="0.4 0.4 0.4 1.0"/>
  </material>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="0.9 0.9 0.9 1.0"/>
  </material>
  <material name="yellow">
    <color rgba="0.8 0.8 0.0 1.0"/>
  </material>
  <material name="black">
    <color rgba="0.15 0.15 0.15 1.0"/>
  </material>
  <!--    Only base link contains the collision bodies-->
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.25"/>
      <mass value="1.0"/>
      <inertia ixx="0.048966669" ixy="0" ixz="0" iyy="0.046466667" iyz="0" izz="0.0041666669"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.25"/>
      <geometry>
        <box size="0.95 0.90 0.5"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="-0.35 0 0.775"/>
      <geometry>
        <box size="0.20 0.90 0.55"/>
      </geometry>
    </collision>
  </link>
  <joint name="base_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="chassis_link"/>
  </joint>
  <link name="chassis_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/body.stl"/>
      </geometry>
      <material name="black"/>
    </visual>
    <inertial>
      <origin rpy="1.5707963267948966 0 1.5707963267948966" xyz="0.012 0.002 0.067"/>
      <mass value="165.304"/>
      <inertia ixx="4.4744" ixy="0.03098" ixz="0.003647" iyy="7.1624" iyz="0.1228" izz="4.6155"/>
    </inertial>
  </link>
  <joint name="right_side_cover_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="chassis_link"/>
    <child link="right_side_cover_link"/>
  </joint>
  <joint name="left_side_cover_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="chassis_link"/>
    <child link="left_side_cover_link"/>
  </joint>
  <link name="left_side_cover_link">
    <visual>
      <origin rpy="0 0 3.141592653589793" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/side-cover.stl"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <link name="right_side_cover_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/side-cover.stl"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <joint name="front_cover_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="chassis_link"/>
    <child link="front_cover_link"/>
  </joint>
  <link name="front_cover_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/end-cover.stl"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="rear_cover_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="chassis_link"/>
    <child link="rear_cover_link"/>
  </joint>
  <link name="rear_cover_link">
    <visual>
      <origin rpy="0 0 3.141592653589793" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/end-cover.stl"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="front_lights_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="chassis_link"/>
    <child link="front_lights_link"/>
  </joint>
  <joint name="rear_lights_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="chassis_link"/>
    <child link="rear_lights_link"/>
  </joint>
  <link name="front_lights_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/lights.stl"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <link name="rear_lights_link">
    <visual>
      <origin rpy="0 0 3.141592653589793" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/lights.stl"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>
  <joint name="top_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="chassis_link"/>
    <child link="top_link"/>
  </joint>
  <link name="top_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/top.stl"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <joint name="axle_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.05"/>
    <parent link="chassis_link"/>
    <child link="axle_link"/>
  </joint>
  <link name="axle_link">
    <visual>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/ridgeback/ridgeback_description/meshes/axle.stl"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>
  <link name="imu_link">
    <inertial>
      <mass value="0.001"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1e-09" ixy="0.0" ixz="0.0" iyy="1e-09" iyz="0.0" izz="1e-09"/>
    </inertial>
  </link>
  <joint name="imu_joint" type="fixed">
    <parent link="chassis_link"/>
    <child link="imu_link"/>
    <origin rpy="0 0 0" xyz="0.2085 -0.2902 0.1681"/>
  </joint>
  <link name="mid_mount"/>
  <joint name="mid_mount_joint" type="fixed">
    <parent link="base_link"/>
    <child link="mid_mount"/>
    <origin rpy="0 0 0" xyz="0 0 0.28"/>
  </joint>
  <!-- Sensor mount -->
  <link name="ridgeback_sensor_mount_link">
    <visual>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_manipulation/data/meshes/ridgeback_sensor_mount.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="ridgeback_sensor_mount_joint" type="fixed">
    <origin rpy="0 0 1.571" xyz="0 0 0"/>
    <parent link="mid_mount"/>
    <child link="ridgeback_sensor_mount_link"/>
  </joint>
  <link name="reference_link"/>
  <joint name="reference_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.343 0 0.77"/>
    <parent link="mid_mount"/>
    <child link="reference_link"/>
  </joint>
  <!--Mount panda on ridgeback-->
  <joint name="arm_mount_joint" type="fixed">
    <parent link="mid_mount"/>
    <child link="franka_mount_link"/>
    <!--        <origin xyz="$(optenv RIDGEBACK_FRANKA_XYZ 0.21 0.0 0.2)" rpy="$(optenv RIDGEBACK_FRANKA_RPY 0 0 0)"/>-->
    <origin rpy="0 0 0" xyz="0.30 0.0 0.445"/>
  </joint>
  <link name="franka_mount_link"/>
  <joint name="panda_joint_franka_mount_link" type="fixed">
    <parent link="franka_mount_link"/>
    <child link="panda_link0"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/</robotNamespace>
      <robotParam>/</robotParam>
    </plugin>
  </gazebo>
  <gazebo reference="panda_link0"/>
  <gazebo reference="panda_link1"/>
  <gazebo reference="panda_link2"/>
  <gazebo reference="panda_link3"/>
  <gazebo reference="panda_link4"/>
  <gazebo reference="panda_link5"/>
  <gazebo reference="panda_link6"/>
  <gazebo reference="panda_link7"/>
  <gazebo reference="panda_link8"/>
  <transmission name="panda_joint1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_joint1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_joint1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="panda_joint2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_joint2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_joint2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="panda_joint3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_joint3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_joint3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="panda_joint4_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_joint4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_joint4_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="panda_joint5_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_joint5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_joint5_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="panda_joint6_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_joint6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_joint6_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="panda_joint7_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_joint7">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_joint7_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="panda_link0">
    <inertial>
      <mass value="2.92"/>
      <origin rpy="0 0 0" xyz="-0.025566 -2.88e-05 0.057332"/>
      <inertia ixx="0.00782229414331" ixy="-1.56191622996e-05" ixz="-0.00126005738123" iyy="0.0109027971813" iyz="1.08233858202e-05" izz="0.0102355503949"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/link0.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/collision/link0.obj"/>
      </geometry>
    </collision>
  </link>
  <link name="panda_link1">
    <inertial>
      <mass value="2.74"/>
      <origin rpy="0 0 0" xyz="0 -0.0324958 -0.0675818"/>
      <inertia ixx="0.0180416958283" ixy="0.0" ixz="0.0" iyy="0.0159136071891" iyz="0.0046758424612" izz="0.00620690827127"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/link1.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/collision/link1.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint1" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="0 0 0" xyz="0 0 0.333"/>
    <parent link="panda_link0"/>
    <child link="panda_link1"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-20" upper="20" velocity="2.1750"/>
  </joint>
  <link name="panda_link2">
    <inertial>
      <mass value="2.74"/>
      <origin rpy="0 0 0" xyz="0 -0.06861 0.0322285"/>
      <inertia ixx="0.0182856182281" ixy="0.0" ixz="0.0" iyy="0.00621358421175" iyz="-0.00472844221905" izz="0.0161514346309"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/link2.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/collision/link2.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint2" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-1.7628" soft_upper_limit="1.7628"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="panda_link1"/>
    <child link="panda_link2"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-20" upper="20" velocity="2.1750"/>
  </joint>
  <link name="panda_link3">
    <inertial>
      <mass value="2.38"/>
      <origin rpy="0 0 0" xyz="0.0469893 0.0316374 -0.031704"/>
      <inertia ixx="0.00771376630908" ixy="-0.00248490625138" ixz="-0.00332147581033" iyy="0.00989108008727" iyz="-0.00217796151484" izz="0.00811723558464"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/link3.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/collision/link3.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint3" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.57079632679 0 0" xyz="0 -0.316 0"/>
    <parent link="panda_link2"/>
    <child link="panda_link3"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-20" upper="20" velocity="2.1750"/>
  </joint>
  <link name="panda_link4">
    <inertial>
      <mass value="2.38"/>
      <origin rpy="0 0 0" xyz="-0.0360446 0.0336853 0.031882"/>
      <inertia ixx="0.00799663881132" ixy="0.00347095570217" ixz="-0.00241222942995" iyy="0.00825390705278" iyz="0.00235774044121" izz="0.0102515004345"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/link4.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/collision/link4.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint4" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.0718" soft_upper_limit="0.0698"/>
    <!-- soft_upper_limitr should be -0.0698, but that brakes simulation when initial position for this joint is 0.0 -->
    <origin rpy="1.57079632679 0 0" xyz="0.0825 0 0"/>
    <parent link="panda_link3"/>
    <child link="panda_link4"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-20" upper="20" velocity="2.1750"/>
  </joint>
  <link name="panda_link5">
    <inertial>
      <mass value="2.74"/>
      <origin rpy="0 0 0" xyz="0 0.0610427 -0.104176"/>
      <inertia ixx="0.030371374513" ixy="6.50283587108e-07" ixz="-1.05129179916e-05" iyy="0.0288752887402" iyz="-0.00775653445787" izz="0.00444134056164"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/link5.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/collision/link5.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint5" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="-1.57079632679 0 0" xyz="-0.0825 0.384 0"/>
    <parent link="panda_link4"/>
    <child link="panda_link5"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-20" upper="20" velocity="2.6100"/>
  </joint>
  <link name="panda_link6">
    <inertial>
      <mass value="1.55"/>
      <origin rpy="0 0 0" xyz="0.0510509 0.009108 0.0106343"/>
      <inertia ixx="0.00303336450376" ixy="-0.000437276865508" ixz="0.000629257294877" iyy="0.00404479911567" iyz="0.000130472021025" izz="0.00558234286039"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/link6.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/collision/link6.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint6" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-0.0175" soft_upper_limit="3.7525"/>
    <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="panda_link5"/>
    <child link="panda_link6"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-20" upper="20" velocity="2.6100"/>
  </joint>
  <link name="panda_link7">
    <inertial>
      <mass value="0.54"/>
      <origin rpy="0 0 0" xyz="0.0109695 0.0107965 0.0650411"/>
      <inertia ixx="0.000888868887021" ixy="-0.00012239074652" ixz="3.98699829666e-05" iyy="0.000888001373233" iyz="-9.33825115206e-05" izz="0.0007176834609"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/link7.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/collision/link7.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint7" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.57079632679 0 0" xyz="0.088 0 0"/>
    <parent link="panda_link6"/>
    <child link="panda_link7"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-20" upper="20" velocity="2.6100"/>
  </joint>
  <link name="panda_link8">
    <inertial>
      <mass value="0.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <joint name="panda_joint8" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.107"/>
    <parent link="panda_link7"/>
    <child link="panda_link8"/>
    <axis xyz="0 0 0"/>
  </joint>
  <joint name="panda_hand_joint" type="fixed">
    <parent link="panda_link8"/>
    <child link="panda_hand"/>
    <origin rpy="0 0 -0.785398163397" xyz="0 0 0"/>
  </joint>
  <link name="panda_hand">
    <inertial>
      <mass value="0.73"/>
      <origin rpy="0 0 0" xyz="0 0.0015244 0.0275912"/>
      <inertia ixx="0.00278560230025" ixy="0.0" ixz="0.0" iyy="0.000400033405336" iyz="0.0" izz="0.00256378041832"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/hand.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <!--origin rpy="0 0 0" xyz="0 0 0"/-->
      <origin rpy="0 0 0" xyz="0 0 0.0333"/>
      <geometry>
        <!--mesh filename="${resources_prefix}${resources_directory}/resources/panda/meshes/collision/hand.${collision_format}"/!-->
        <box size="0.035 0.19 0.065"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="panda_hand"/>
  <gazebo reference="panda_leftfinger"/>
  <gazebo reference="panda_rightfinger"/>
  <transmission name="panda_finger_joint1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_finger_joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_finger_joint1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="panda_finger_joint2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="panda_finger_joint2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="panda_finger_joint2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="panda_leftfinger">
    <inertial>
      <mass value="0.1"/>
      <origin rpy="0 0 0" xyz="0 0.0145644 0.0227941"/>
      <inertia ixx="3.01220925051e-05" ixy="0.0" ixz="0.0" iyy="2.95873808038e-05" iyz="0.0" izz="6.95125211657e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/finger.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <!--origin rpy="0 0 0" xyz="0 0 0"/-->
      <origin rpy="0 0 3.14159265359" xyz="0 0.01 0.028"/>
      <geometry>
        <!--mesh filename="${resources_prefix}${resources_directory}/resources/panda/meshes/collision/finger.${collision_format}"/-->
        <box size="0.02 0.02 0.05"/>
      </geometry>
    </collision>
  </link>
  <link name="panda_rightfinger">
    <inertial>
      <mass value="0.1"/>
      <origin rpy="0 0 3.14159265359" xyz="0 0.0145644 0.0227941"/>
      <inertia ixx="3.01220925051e-05" ixy="0.0" ixz="0.0" iyy="2.95873808038e-05" iyz="0.0" izz="6.95125211657e-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
      <geometry>
        <mesh filename="/home/giuseppe/catkin_ws/src/sampling_based_control/mppi_examples/mppi_panda/resources/panda/meshes/visual/finger.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <!-- origin rpy="0 0 3.14159265359" xyz="0 0 0"/ -->
      <origin rpy="0 0 3.14159265359" xyz="0 -0.01 0.028"/>
      <geometry>
        <!--mesh filename="${resources_prefix}${resources_directory}/resources/panda/meshes/collision/finger.${collision_format}"/-->
        <box size="0.02 0.02 0.05"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_finger_joint1" type="prismatic">
    <parent link="panda_hand"/>
    <child link="panda_leftfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 1 0"/>
    <limit effort="20" lower="-0.001" upper="0.04" velocity="0.3"/>
  </joint>
  <joint name="panda_finger_joint2" type="prismatic">
    <parent link="panda_hand"/>
    <child link="panda_rightfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 -1 0"/>
    <limit effort="20" lower="-0.001" upper="0.04" velocity="0.3"/>
  </joint>
  <link name="panda_grasp"/>
  <joint name="panda_grasp_joint" type="fixed">
    <parent link="panda_hand"/>
    <child link="panda_grasp"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 0.1"/>
  </joint>
</robot>

